### PR TITLE
[BugFix] fix inspect meta functions (#32710)

### PR DESCRIPTION
**Pull request by murphyatwork**
Fixes #issue

1. Support both internal table and external table for `inspect_related_mv function`
2. Add `inspect_mv_relationships` function
3. Report root cause error message
4. Try to fix mv relationship when active mv
5. Fix the potential memory leak in `ConnectorTblMetaInfoMgr`

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [x] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 6 / 7 (85.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java|5|6|83.33%|[1189]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorEvaluator.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 6 / 7 (85.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java|5|6|83.33%|[1189]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorEvaluator.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 6 / 7 (85.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java|5|6|83.33%|[1189]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorEvaluator.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 9 / 29 (31.03%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ConnectorTableInfo.java|0|1|00.00%|[55]|
|:large_blue_circle:|com/starrocks/connector/ConnectorTblMetaInfoMgr.java|0|10|00.00%|[111, 112, 113, 114, 115, 116, 117, 118, 119, 120]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java|5|14|35.71%|[1189, 1220, 1222, 1224, 1225, 1226, 1227, 1230, 1231]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorEvaluator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/AlterMVJobExecutor.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 32 / 37 (86.49%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java|9|14|64.29%|[1189, 1224, 1225, 1226, 1227]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorTableInfo.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorTblMetaInfoMgr.java|16|16|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorEvaluator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/AlterMVJobExecutor.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 32 / 37 (86.49%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java|9|14|64.29%|[1189, 1224, 1225, 1226, 1227]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorTableInfo.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorTblMetaInfoMgr.java|16|16|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorEvaluator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/AlterMVJobExecutor.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java [hunk 1]
```diff
@@ -1202,8 +1208,27 @@ public static ConstantOperator inspect_related_mv(ConstantOperator name) {
...
-            dbTable.getLeft().readUnlock();
+            mayDb.ifPresent(Database::readUnlock);
+        }
+    }
+
+    /**
+     * Return the content in ConnectorTblMetaInfoMgr, which contains mapping information from base table to mv
+     */
+    @ConstantFunction(name = "inspect_mv_relationships", argTypes = {}, returnType = VARCHAR, isMetaFunction = true)
+    public static ConstantOperator inspect_mv_relationships() throws AccessDeniedException {
```
Comment (satanson) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java:1219]: use inspectMvRelationships instead ?

Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java:1219]: fixed

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 32 / 37 (86.49%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java|9|14|64.29%|[1189, 1224, 1225, 1226, 1227]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorTableInfo.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorTblMetaInfoMgr.java|16|16|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorEvaluator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/AlterMVJobExecutor.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by satanson — APPROVED** [review]

**Review by LiShuMing — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**